### ISSUE TITLE
Support controlling separator when appending current value using `Set Suite Metadata`, `Set Test Documentation` and other such keywords

### ISSUE THREAD

**Issue by Andyc545** [robotframework/robotframework#5215]
Two deficiencies identified with the current Set Suite Metadata keyword, which we use to record stuff like version numbers.

- [ ] Add argument to define a separator between multiple metadata returned values when append=True. Currently, the default behavior is just to put in a space. 

https://robotframework.org/robotframework/latest/libraries/BuiltIn.html#Set%20Suite%20Metadata
> Arguments:
> name
> value
> append = False
> top = False
> 

### Example
**current:** 
```
    FOR    ${device}    IN   @{devices}
        Set Suite Metadata | Versions | ${version} | append=True
    END
```
**output** 
Versions: 1.0 1.2 1.0 1.0 2.3

**proposal:**
```
    FOR    ${device}    IN   @{devices}
        Set Suite Metadata | Versions | ${version} | append=True  | separator=,
    END
```
**output** 
Versions: 1.0, 1.2, 1.0, 1.0, 2.3

- [ ]  Argument to allow unique returned values only
To continue the example above:
```
    FOR    ${device}    IN   @{devices}
        Set Suite Metadata | Versions | ${version} | append=True  | separator=, | unique=True
    END
```
**output** 
Versions: 1.0, 1.2, 2.3

**Comment by pekkaklarck** [comment]
Making the separator configurable would be easy. To allow full configuration, the separator should replace the current space, so you needed to use `separator=,${SPACE}` to get items separated with a comma and a space.

Metadata values are just strings, and it wouldn't be possible to handle uniqueness properly. For example, if the current value would be `2.0, 7.0, 11.0`, it would contain `1.0` and it wouldn't be added. Trying to parse the value more intelligently would be harder and there nevertheless could be false results. I believe it's better that you use an `IF` block if you want something to be added conditionally.

An alternative approach for all this would be adding values to a list, removing possible duplicates, joining list items together, and then using `Set Suite Metadata` just once with the final value. That said, adding `separator` would help in simple cases and be a logic extension for the current `append` functionality.

Are you, or anyone else, interested to implement the `separator` functionality?

**Comment by pekkaklarck** [comment]
I noticed that there are various other keywords that support appending to the current value and believe they should all be enhanced. The keywords are `Set Test Message`, `Set Test Documentation`, `Set Suite Documentation` and `Set Suite Metadata` and they all got `append` in RF 2.7.7 ( #1348).

**Comment by pekkaklarck** [comment]
Great! I'll add this to RF 7.2 scope then.

If you haven't contributed to this project earlier, I recommend reading the contributing guidelines and instructions how to create acceptance test in `atest/README.rst`. Implementation itself ought to be pretty easy, there's a helper method that all affected keywords use, but there are few other things to take into account:
1. Acceptance tests. I believe each keyword should get one.
2. Documentation update. This includes a note about the feature being new in RF 7.2. You can see other keywords with such notes as examples.

**Comment by pekkaklarck** [comment]
Implemented in PR #5221 by @m-usmani. Thanks for a great contribution!
